### DISCUSSION TITLE
[Feature Request]: Add an option to not modify zshrc file

### DISCUSSION THREAD

**Discussion by donferi** [beyondcode/herd-community#360] [Ideas]
### Feature Description

I use `nix` on macOS and my `.zshrc` file is configured through nix. Herd writes to it and now my config is no longer "pure" and nix overwrites it next time I run it.

It would be nice to have an option not to add this by default and just print the lines I need to add to my config.

### Is this feature valuable for other users as well and why?

Yes, anyone using nix or really anyone that splits their `.zshrc`, like using a shared `dotfiles` in multiple computer but you may have herd in only one. In those scenarios you normally have a `.localrc` or similar.

**Comment by mpociot** [comment]
Could you tell me how nix manages your .zshrc file? Is it similar to a symlink?

**Comment by donferi** [comment]
> Could you tell me how nix manages your .zshrc file? Is it similar to a symlink?

@mpociot  Correct, its a symlink. With my regular user I can't modify it, its ready only.

However after herd runs it seems like the symlink is removed and its modified to inject herd to the path.

Eg before herd:
```
❯ ls -la ~/.zshrc
lrwxr-xr-x - donferi 13 Dec 18:17 /Users/donferi/.zshrc -> /nix/store/nvq7f5iv7hnvzaskzqqqab18lfcahhi3-home-manager-files/.zshrc
```

After herd runs:
```
❯ ls -la ~/.zshrc
.rw-r--r--@ 3.9k donferi 13 Dec 18:18 /Users/donferi/.zshrc
```

**Reply by eleftrik** [reply]
@donferi Same problem here. IMO Herd shouldn't overwrite .zshrc.

**Comment by punkfairie** [comment]
I would love this as well. I split my .zshrc into multiple files by language (a la [holman's topic-based dotfiles](https://github.com/holman/dotfiles)) and I would MUCH rather be able to do this manually. Even just having an option during install to add the lines automatically or not, so the user could choose.

**Comment by SolidRhino** [comment]
I use nix as well and love this to be available.

**Comment by rynhndrcksn** [comment]
While I don't use nix, I would also love for this to be an option.
